flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/AbstractQueryableStateTestBase.java

```diff
@@ -359,12 +355,12 @@ public Integer getKey(Tuple2<Integer, Long> value) throws Exception {
 		} finally {
 			// Free cluster resources
 			if (jobId != null) {
-				scala.concurrent.Future<CancellationSuccess> cancellation = cluster
-						.getLeaderGateway(deadline.timeLeft())
+				cluster.getLeaderGateway(deadline.timeLeft())
 						.ask(new JobManagerMessages.CancelJob(jobId), deadline.timeLeft())
-						.mapTo(ClassTag$.MODULE$.<JobManagerMessages.CancellationSuccess>apply(JobManagerMessages.CancellationSuccess.class));
+						.mapTo(ClassTag$.MODULE$.<CancellationSuccess>apply(CancellationSuccess.class));
 
-				Await.ready(cancellation, deadline.timeLeft());
+				// we are not waiting for the cancellation to happen because the
+				// job has actually failed, as tested above.
```

Comment:
you can't guarantee this in the `finally` block. (for example if submitJobDetached failed but the job is actually running)

Reply:
You are right. So I will just remove the `finally` block and check if the status of the job is `FAILED`.